tests/FakeItEasy.Tests/ArgumentConstraintTestBase.of.T.cs

```diff
@@ -14,5 +18,7 @@ protected ArgumentConstraintTestBase()
         }
 
         protected abstract void CreateConstraint(IArgumentConstraintManager<T> scope);
+        public abstract IEnumerable<object[]> ValidValues();
+        public abstract IEnumerable<object[]> InvalidValues();
     }
 }

```

Comment:
I used ` ArgumentConstraintTestBase<T>` as the test context base, because it was convenient. If this were a serious proposal, we'd probably pick something better.